conn.go

```diff
@@ -430,6 +430,11 @@ func (conn *Conn) checkIO(err error) {
 		return
 	}
 
+	if err == io.EOF {
```

Comment:
perhaps:
```suggestion
	if err == io.EOF || xerrors.Is(err, io.EOF) {
```

to future-proof?

also: what about `io.ErrUnexpectedEOF` ?

Reply:
`io.ErrUnexpectedEOF` probably means "Protocol Error", at least with TCP. I wasn't sure if it's appropriate to treat this as a hard fault because the next read could be fine.
For some protocols it could very well indicate connection closure but the next read would return `io.EOF`. It's no different from `rw.Read() == nil` and then the connection closes immediately after that.

Reply:
ok.